greeter/README.md

```diff
@@ -429,46 +568,50 @@ You will see message `'你好, name'` printed instead of `'name, 你好'`.
 Related documentation:
 [Configuration by convention](https://loopback.io/doc/en/lb4/Context.html#configuration-by-convention)
 
-## Part 3: Enable Caching
+## Part 4: Enable Caching
 
 We have showed in the previous steps that how to decouple your application and design it as a microservice. To enhance the idea of building a microservice, we will add more services to the greeting app.
 
 To reduce the cost of calculation, we'd like to use a caching service for the greeting application. We will show how to bind a caching service, interceptor and observer to the app.
 
 The caching service will cache the greeting messages for a certain amount of time. And cached messages will be swept out when they are expired.
 
+![Greeting application diagram](images/greeting-app-overview.png)
+
 Again, we are using `bind` and `injection` to bind our app with an Interceptor and Observer:
 
 - Interceptors: are reusable functions that provide [aspect-oriented](https://en.wikipedia.org/wiki/Aspect-oriented_programming) logic around method invocations.
 - Observer: is a LoopBack application that has its own life cycles at runtime. An [`Observer allows`](https://loopback.io/doc/en/lb4/Life-cycle.html#the-lifecycleobserver-interface) artifacts to participate in the life cycles. The transition of states is triggered by two functions: `start()` and `stop()`.
 
 ### Step 1: Create a caching service
 
-1.  Copy the contents of `caching.service.ts` to a new file in your workspace named `src/caching.service.ts`.
+Stop the app with Ctrl+C.
+
+1.  Copy the contents of [caching.service.ts](https://github.com/strongloop/cascon2019/blob/dremond_update/greeter/src/caching.service.ts) to a new file in your workspace named `src/caching.service.ts`.
```

Comment:
idk if the link would work. And if the instruction will be printed, I think the path is enough.

Reply:
If you don't have this link, how will the user get the contents of caching.service.ts ?

Reply:
I think we can keep the full details in git repository, and trim the content for the printed version, (and there will be more stuff to take out than the links :)